File: Android/VideoStreamer/app/src/main/java/com/omercankoc/videostreamer/VideoStreamActivity.java
package com.omercankoc.videostreamer;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.net.Socket;

public class VideoStreamActivity extends AppCompatActivity {

    Thread thread1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_stream);

        thread1 = new Thread(new Thread1());
        thread1.start();
    }

    private PrintWriter output;
    private BufferedReader input;
    class Thread1 implements Runnable {
        @Override
        public void run() {
            Socket socket;
            try {
                socket = new Socket("172.24.1.1", 8080);
                output = new PrintWriter(socket.getOutputStream());
                input = new BufferedReader(new InputStreamReader(socket.getInputStream()));
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(getApplicationContext(),"Connected!",Toast.LENGTH_LONG).show();
                    }
                });
                new Thread(new Thread2()).start();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
    class Thread2 implements Runnable {
        @Override
        public void run() {
            while (true) {
                try {
                    final String message = input.readLine();
                    if (message != null) {
                        runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                System.out.println(message);
                            }
                        });
                    } else {
                        thread1 = new Thread(new Thread1());
                        thread1.start();
                        return;
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: VideoStream/app/src/main/java/com/carvio/videostream/VideoStreamActivity.kt
package com.carvio.videostream

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import io.ktor.network.selector.*
import io.ktor.network.sockets.*
import io.ktor.util.Identity.decode
import io.ktor.utils.io.*
import kotlinx.coroutines.*
import kotlinx.coroutines.Dispatchers.IO
import java.io.*
import java.net.Socket
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.util.*


class VideoStreamActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video_stream)

        // UTF8
        // SETTER FRAME ARRAY
        // FIND
        // CREATE FRAME
        // IMDECODE

        var frame = String()

        GlobalScope.async {
            try {

                val socket = Socket("172.24.1.1", 8080)
                var inputStreamReader : InputStreamReader = InputStreamReader(socket.getInputStream(),StandardCharsets.UTF_8)
                var bufferedReader : BufferedReader = BufferedReader(inputStreamReader,)

                while (true){

                    var stream = bufferedReader.readLine()
                }

            } catch (e: IOException){
                e.printStackTrace()
            }
        }
    }
    companion object {
        val TAG = VideoStreamActivity::class.simpleName
    }
}